README.md

```diff
@@ -1,17 +1,110 @@
 
-## Compatibility
-To keep the SDK compatible with Java, refer to the following docs from Kotlin:
-* [Calling Kotlin from Java](https://kotlinlang.org/docs/java-to-kotlin-interop.html)
-* [Mixing Java and Kotlin in one project](https://kotlinlang.org/docs/mixing-java-kotlin-intellij.html)
-* [Using Java records in Kotlin](https://kotlinlang.org/docs/jvm-records.html)
+## ✨ Features
+
+- 🔌 **Modular architecture** (`sdk-core`, `sdk-api`)
+- 🔄 Generic HTTP client with support for **GET** and **POST**
+- 🔐 Pluggable authentication (JWT, Basic Auth, or custom)
+- ✅ Built-in support for **error handling** and **response validation**
+- 📦 Easily publishable to GitHub Packages
+- 🚀 Ready-to-use SDK factory for simplified instantiation
+- 📘 Example integration using [PokéAPI](https://pokeapi.co)
+
+## 📁 Structure Overview
+
+```plaintext
+sdk-template/
+├── sdk-core/                      # Core utilities and infrastructure
+│   └── src/main/kotlin/io/etip/sdk/core/
+│       ├── BaseHttpClient.kt       # Generic HTTP client with GET/POST support
+│       ├── HttpClientFactory.kt    # Factory responsible for creating and configuring instances of `HttpClient`.
+│       ├── exceptions/             # Common SDK exceptions
+│       └── routes/                 # Sealed route definitions
+│
+├── sdk-api/                       # API-specific integrations
+│   └── src/main/kotlin/io/etip/sdk/api/
+│       ├── config/                 # ClientConfig, Environment enum
+│       ├── integration/            # ApiService interface and implementation
+│       └── factory/                # SdkFactory to instantiate the API client
+│
+├── examples/                      # Example usage of the SDK
+│   └── basic-example/             # Pokémon API usage demo
+│
+├── .github/workflows/             # CI/CD workflows for release
+│   └── release.yml
+│
+└── README.md                      # Project overview and usage guide
+```
+
+### 🔐 Authentication Strategy
+
+The SDK follows a flexible authentication strategy that separates concerns between `sdk-core` and `sdk-api`.
+
+#### In `sdk-core`
+
+Authentication must be configured at the `HttpClient` level and is agnostic to the specific authentication mechanism. Whether the integration requires **Basic Auth**, **JWT Bearer tokens**, or any other custom scheme, the core does **not** hard-code any implementation. Instead, it provides the infrastructure (like `BaseHttpClient`) to accept headers, allowing the `sdk-api` layer to inject credentials dynamically.
+
+You can also implement custom authenticators using a strategy pattern by defining an abstract `Authenticator` interface in `sdk-core`. The concrete implementation (e.g., `JwtAuthenticator`, `BasicAuthenticator`) would live in `sdk-api`.
```

Comment:
Very confused about the core and api relatioin here.

Reply:
See [here](https://github.com/eTipio/client-sdk-kotlin-template/pull/20#discussion_r2121439579).